Upgrade to opentelemetry 0.25

ymgyt opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests.

Contributors guide: https://github.com/tokio-rs/tracing-opentelemetry/blob/master/CONTRIBUTING.md
-->

## Motivation

<!--
Explain the context and why you're making that change. What is the problem
you're trying to solve? If a new feature is being added, describe the intended
use case that feature fulfills.
-->

Update opentelemetry depencencies to 0.25
https://github.com/open-telemetry/opentelemetry-rust/releases/tag/opentelemetry-0.25.0

From version 0.25, OpenTelemetry started using a unified version across all crates.
https://github.com/open-telemetry/opentelemetry-rust/issues/2084

## Solution

<!--
Summarize the solution and provide any necessary context needed to understand
the code change.
-->

* Upgrade opentelemetry deps to 0.25
* Remove `AttributeSet` from tests
   * https://github.com/open-telemetry/opentelemetry-rust/pull/2045
* Rename deprecated `deployment.environment` semantic conv to `deployment.environment.name`
  * https://opentelemetry.io/docs/specs/semconv/attributes-registry/enduser/

ymgyt commented on tests/metrics_publishing.rs:
```diff
@@ -702,3 +675,16 @@ where
         Ok(())
     }
 }
+
```
I don’t understand the issue yet, but the test failed because the order of attributes in the exported metrics differed each time. Therefore, I added a sorting process during the assertion.

ymgyt commented on src/layer.rs:
```diff
@@ -1693,7 +1693,7 @@ mod tests {
 
             // This can cause a deadlock if `on_record` locks extensions while attributes are visited
-            span.record("exception", &tracing::field::debug(&context));
+            span.record("exception", tracing::field::debug(&context));
```
`clippy::needless_borrows_for_generic_args`

ymgyt commented on src/layer.rs:
```diff
@@ -646,7 +646,7 @@ where
     ///
     /// * Only events without a message field (unnamed events) and at least one field with the name error
-    /// are considered for mapping.
+    ///   are considered for mapping.
```
`doc_lazy_continuation`

djc reviewed (commented):
Thanks for working on this! I think this is mostly looking good, but I'd like to have a cleaner commit history where basically each commit could pass CI (though I won't ask you to actually test this). As such:

* Please have reorder the commits to put the clippy commit first, as I assume this is independent of the opentelemtry upgrade
* Please squash the commit with the dependency bump with the commit that fixes the tests, also include the commit for fixing the warnings from deprecations
* Add a version bump for this crate into the commit that updates the CHANGELOG

ymgyt commented:
Thank you for your review.
I first made a commit addressing Clippy warnings.
I squashed the commits for the OpenTelemetry version bump and deprecation fixes.
I also updated the changelog and bumped the crate version.

djc commented:
* Published tracing-opentelemetry v0.26.0 at registry `crates-io`
* [new tag]         v0.26.0 -> v0.26.0
* [Release notes](https://github.com/tokio-rs/tracing-opentelemetry/releases/tag/v0.26.0)

ymgyt commented:
@djc 
Sorry for the issue with the CHANGELOG update not being as intended. In the PR, I mistakenly created it as:

> # [Unreleased]

but it should have been:

> # 0.26.0 (September 10, 2024).
Should I create a PR to address this now?

https://github.com/tokio-rs/tracing-opentelemetry/blob/v0.26.0/CHANGELOG.md

djc commented:
Yes, please.
